xrootd/client/session.go

```diff
@@ -163,88 +162,91 @@ func (sess *session) Close() error {
 	return nil
 }
 
+// handleReadError handles an error encountered while reading and parsing a response.
+// If the current session is equal to the initial, the error is considered critical and handleReadError panics.
+// Otherwise, the current session is closed and all requests are redirected to the initial session.
+// See http://xrootd.org/doc/dev45/XRdv310.pdf, p. 11 for details.
+func (sess *session) handleReadError(err error) {
```

Comment:
according to the coverage report, it seems `handleReadError` isn't exercized in our tests.
could we fix this?